storage.go

```diff
@@ -285,15 +283,31 @@ func (s *Storage) Unlock(key string) error {
 	}
 
 	lockKey := fmt.Sprintf("LOCK-%s", key)
-
-	return s.Delete(lockKey)
+	it, rev, err := s.getItem(lockKey)
+	if err != nil && !errors.Contains(err, errNotExist) {
+		return err
+	}
+	// if lock doesn't exist or is empty, break to create a new one
+	if isEmpty(it.Contents) {
+		return nil
+	}
+	it.PrimaryKey = lockKey
+	it.Contents = emptyRegistryEntry[:]
+	it.LastUpdated = time.Now().UTC()
+	bytes, err := json.Marshal(it)
+	if err != nil {
+		return errors.AddContext(err, "failed to marshal the item record")
+	}
+	dataKey := crypto.HashBytes([]byte(it.PrimaryKey))
+	return s.SkyDB.Write(bytes, dataKey, rev+1)
 }
 
 // getItem fetches an ItemRecord from SkyDB.
 func (s *Storage) getItem(key string) (Item, uint64, error) {
 	dataKey := crypto.HashBytes([]byte(key))
 	data, rev, err := s.SkyDB.Read(dataKey)
-	if err != nil && errors.Contains(err, skydb.ErrNotFound) {
+	// The string check is annoying and probably unnecessary but I want to get this working.
+	if err != nil && (errors.Contains(err, skydb.ErrNotFound) || strings.Contains(err.Error(), "registry entry not found")) {
```

Comment:
same remark about the error string comparisons

Reply:
This one is even worse, tbh. because there are two - `registry entry not found` and `registry entry not found withing given time`. And both mean the same thing to me.

I just saw that there are exposed errors. I will use those. I don't think that's any improvement in readability or dependability of the software but it's the best I can do to avoid string comparisons.